### DISCUSSION TITLE
how to use an MPV script

### DISCUSSION THREAD

**Discussion by jarss** [iina/iina#4605] [General]
In Linux, I used [mpv-video-splice](https://github.com/pvpscript/mpv-video-splice/blob/master/README.md), which allows you to cut parts of a video as you watch and save as individual files. I don't know how to get it to work in iina. 

I have the settings configured to allow MPV configurations in ~/.config/mpv, and I have the Lua files there, as instructed in the splice readme. The shortcuts for it do not work, I just get the "error beep" when I try to use it with alt+t or alt+c, for example.

**Comment by low-batt** [comment]
@svobs is more knowledgeable than me in this area of IINA. Here is my take on what the problem is…

The sound you are hearing is probably `Boop`. That is the default macOS `Alert sound`. It is configurable in the macOS `System Settings` in the `Sound` section.

The reason why the key press is not being recognized is because the key bindings are being established by the [mpv-video-splice](https://github.com/pvpscript/mpv-video-splice/tree/master) Lua script dynamically at runtime using the mpv mp function [mp.add_key_binding](https://mpv.io/manual/stable/#lua-scripting-[,flags]]) as seen [here](https://github.com/pvpscript/mpv-video-splice/blob/53fbd794ad101d25e2eaeeb3850ab037e9067006/mpv-splice.lua#L326-L330):
```lua
mp.add_key_binding('Alt+t', "put_time", put_time)
mp.add_key_binding('Alt+p', "show_times", show_times)
mp.add_key_binding('Alt+c', "process_video", process_video)
mp.add_key_binding('Alt+r', "reset_current_slice", reset_current_slice)
mp.add_key_binding('Alt+d', "delete_slice", delete_slice)
```

There are some other calls to `mp.add_key_binding` as well.

IINA needs to know when Lua scripts dynamically establish key bindings. Normally `libmpv` clients are informed of such changes through [events](https://mpv.io/manual/master/#list-of-events). However in this case mpv is not emitting an event IINA can listen for to know about this new key binding. The open mpv issue https://github.com/mpv-player/mpv/issues/10275 requests a `MPV_EVENT_PROPERTY_CHANGE` event be emitted in this case.

To workaround this you will need to manually map keys to the mpv [script-binding](https://mpv.io/manual/stable/#command-interface-script-binding) command in the `Key Bindings` section of IINA settings. Search the `mpv Default` key bindings for that command for examples.
